docs/content/standards/kiosk-apps/nft-rental.mdx

```diff
@@ -0,0 +1,418 @@
+- Reclaim for the lender
+
+### Structs
+
+The object model of the `nft_rental` module provides the structure of the app, beginning with the `Rentables` object. The struct only has the `drop` ability and acts as the extension key for the Kiosk `Rentables` extension.
+
+```move
+struct Rentables has drop {}
+```
+
+The `Rented` struct represents a rented item. The only field the struct includes is the ID of the object. 
+
+```move
+struct Rented has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Listed` struct represents a listed item. The only field the struct includes is the ID of the object.
+
+```move
+struct Listed has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Promise` struct is created for borrowing by value. The `Promise` operates as the [hot potato](../../concepts/sui-move-concepts/patterns/hot-potato.mdx) that can only be resolved by returning the item back to the extension's `Bag`.
+
+```move
+struct Promise has store {
+  item_id: Rented,
+  duration: u64,
+  start_date: u64,
+  price_per_day: u64,
+  renter: address,
+  borrower_kiosk: ID
+}
+```
+
+The `Rentable` struct is as a wrapper object that holds an asset that is being rented. Contains information relevant to the rental period, cost, and renter.
+
+```move
+struct Rentable<T: key + store> has store {
+      object: T,
+      duration: u64, // total amount of time offered for renting in days
+      start_date: Option<u64>, // initially undefined, is updated once someone rents it
+      price_per_day: u64,
+      kiosk_id: ID // the kiosk id that the object was taken from
+}
+```
+
+The `RentalPolicy` struct is a shared object that every creator mints. Defines the royalties the creator receives from each rent invocation.
+
+```move
+struct RentalPolicy<phantom T> has key, store {
+	id: UID,
+	balance: Balance<SUI>,
+	/// Note: Move does not support float numbers.
+	/// If you need to represent a float, you need to determine the desired precision and use a larger integer representation.
+	/// For example, percentages can be represented using basis points:
+	/// 10000 basis points represents 100% and 100 basis points represents 1%.
+	amount_bp: u64
+}
+```
+
+The `ProtectedTP` object is a shared object that every creator mints, regardless of their intent to enforce royalties. The object provides authorized access to an empty `TransferPolicy`.
+
+```move
+struct ProtectedTP<phantom T> has key, store {
+	id: UID,
+	transfer_policy: TransferPolicy<T>,
+	policy_cap: TransferPolicyCap<T>
+}
+```
+
+### Function signatures
+
+The NFT Rental example includes the following functions that define the project's logic. 
+
+The `install` function enables installation of the `Rentables` extension in a Kiosk.
+
+```move
+public fun install(kiosk: &mut Kiosk, cap: &KioskOwnerCap, ctx: &mut TxContext)
+```
+
+The `remove` function enables the owner (and only the owner) of the kiosk to remove the extension. The extension storage must be empty for the transaction to succeed.
```

Comment:
In the specific case of the renting extension, what does it mean for the extension storage to be empty, and how does one make that happen?

Reply:
It means that you are no longer renting or borrowing any items any more. Adding this as a clarification